flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/stream/StreamExecChangelogNormalize.java

```diff
@@ -81,6 +84,14 @@ public StreamExecChangelogNormalize(
                 tableConfig
                         .getConfiguration()
                         .getBoolean(ExecutionConfigOptions.TABLE_EXEC_MINIBATCH_ENABLED);
+
+        GeneratedRecordEqualiser generatedEqualiser =
+                new EqualiserCodeGenerator(
+                                rowTypeInfo.toRowType().getFields().stream()
+                                        .map(RowType.RowField::getType)
+                                        .toArray(LogicalType[]::new))
```

Comment:
It seems that we have more than 3 places using a long method chain to create the field logical types. I think we can provide a new constructor for EqualiserCodeGenerator to accepting `RowType`. This can simplify code and improve readabliity. E.g. 

```scala
  def this(rowType: RowType) = {
    this(rowType.getChildren.asScala.toArray)
  }
```

Reply:
done with commit https://github.com/apache/flink/pull/14863/commits/748ad82745d9d493922150fe007136e125a50209